workers/shared-worker-parse-error-failure.html

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<title>SharedWorker: parse error failure</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+
+promise_test(async () => {
+  const scriptURL = 'modules/resources/syntax-error.js';
+  const worker = new SharedWorker(scriptURL, { name: 'classic',
+                                               type: 'classic' });
+  return new Promise(resolve => worker.onerror = resolve);
+}, 'Classic shared worker construction for script with syntax error should ' +
+   'dispatch an ErrorEvent.');
+
+promise_test(async () => {
```

Comment:
These tests pass in Firefox because the `import` is a syntax error. Maybe we should add something like this? (I haven't tested this code so it might be wrong.)

```js
// Can put at the top of the test file I think
let supportsModuleWorkers = false;
const options = {
  get type() {
    supportsModuleWorkers = true;
  }
};
new Worker("data:,", options).terminate();

// in each test:
assert_precondition(supportsModuleWorkers, "Supports module workers");
```

Reply:
Thanks for the advice. It's true that we should check if module shared worker is supported, but checking it from if type() is called seems to be not appropriate. In FireFox, module is not supported but WorkerOptions is supported so type() could be called depending on their implementation.

In this test, we only use non-nested static import as a module feature so we can assume module is supported when non-nested static import runs successfully. To check this, the test might timeout when the vendor doesn't support parse error handling, so I think it's better to divide this test into the test for classic and the test for module so that the test for classic runs appropriately even if the test for module timeouts.